packages/@snowpack/plugin-webpack/plugin.js

```diff
@@ -10,8 +11,130 @@ const CopyPlugin = require("copy-webpack-plugin");
 const { JSDOM } = jsdom;
 const cwd = process.cwd();
 
-function insertAfter(newNode, existingNode) {
-  existingNode.parentNode.insertBefore(newNode, existingNode.nextSibling);
+function insertBefore(newNode, existingNode) {
+  existingNode.parentNode.insertBefore(newNode, existingNode);
+}
+
+function parseHTMLFiles({ srcDirectory }) {
+  // Get all html files from the output folder
+  const pattern = srcDirectory + "/**/*.html";
+  const htmlFiles = glob
+    .sync(pattern)
+    .map((htmlPath) => path.relative(srcDirectory, htmlPath));
+
+  const doms = {};
+  const jsEntries = {};
+  for (const htmlFile of htmlFiles) {
+    const dom = new JSDOM(fs.readFileSync(path.join(srcDirectory, htmlFile)));
+
+    //Find all local script, use it as the entrypoint
+    const scripts = Array.from(dom.window.document.querySelectorAll("script"))
+      .filter((el) => el.type.trim().toLowerCase() === "module")
+      .filter((el) => !/^[a-zA-Z]+:\/\//.test(el.src));
+
+    for (const el of scripts) {
+      const src = el.src.trim();
+      const parsedPath = path.parse(src);
+      const name = parsedPath.name;
+      if (!(name in jsEntries)) {
+        jsEntries[name] = {
+          path: path.join(srcDirectory, src),
+          occurrences: [],
+        };
+      }
+      jsEntries[name].occurrences.push({ script: el, dom });
+    }
+
+    doms[htmlFile] = dom;
+  }
+  return { doms, jsEntries };
+}
```

Comment:
👍 I don’t mind the cleanup at all. Thanks for making it easier to follow. Appreciate that you kept most of the code as-is.